### ISSUE TITLE
Issues with nesting classes

### ISSUE THREAD

**Issue by HassanEmam** [TheLartians/modern-wasm-starter#24]
I have created a library that parses a tsv file. Each line parsed is added to an object of another class that is used as a collections.

When I used the template to compile I had some initial issues accessing the filesystem but that has been sorted by modifying the CMAKELists and adding some extra arguments.

Now all compiles ok and I have exposed the functions and members using Glue; I am faced a new issue that I can access a member but when I try to console log the data in that member it does not recognise it. 

The code for the library is https://github.com/Constology/XERNative

Any suggesting on what is the best practice to write the Glue code for this library?

**Comment by TheLartians** [comment]
Hey thanks for the issue! If I understand correctly, you want be be able to `console.log` a C++ object representing your CSV?
In that case you can set the `glue::keys::operators::tostring` property of the Glue class to a function that returns the string value for that object.
 
Unfortunately I don't have the time to write a proper documentation, but you can [find an example in the unit tests](https://github.com/TheLartians/EmGlue/blob/11b154c97376d4bbf6a586ad1d785116ef5f7fd1/test/source/state.cpp#L148).

**Comment by FireBlaze236** [comment]
Hello, I am working on the same problem as here. 
Suppose we have a class called, 
```cpp
class Writer{
public:
    float prop = 1.02f;
};
```
Glued like follows:
```cpp
lib["Writer"] = glue:createClass<Writer>().addConstructor().addMember("prop", &Writer::prop);
```
```js
console.log(readerModule.prop());
```
prints the floating point number.

Now if we had:

```cpp
class Writer{
public:
    LibaryClass prop;
};
```
And then.
```js
console.log(readerModule.prop());
```

This prints Any{}. So, my question is how to actually create the bindings for these classes. Do I have to manually create all the bindings? Or is there a preferred way of creating those mappings.

**Comment by HassanEmam** [comment]
The glue code with the template is published here

https://github.com/Constology/XERNative/tree/wasm

**Comment by TheLartians** [comment]
> This prints Any{}. So, my question is how to actually create the bindings for these classes. Do I have to manually create all the bindings? Or is there a preferred way of creating those mappings.

Yes, you would need to add a definition somewhere for all types that you want to interact with from JavaScript. So in your case, to be able to `console.log` the `LibraryClass`, you would need to add something like

```c++
lib["Writer"] = glue:createClass<Writer>().addConstructor().addMember("prop", &Writer::prop);

lib["LibaryClass"] = glue:createClass<LibaryClass>().addMember(
  glue::keys::operators::tostring,
  [](const LibaryClass &obj){
    return <code to convert LibaryClass to a string>;
  }
);
```

**Comment by TheLartians** [comment]
@HassanEmam @FireBlaze236 did this help with your issue?

**Comment by HassanEmam** [comment]
@TheLartians thanks for your response. I am still struggling to be honest.

I have a Reader class that parses a file which contains Tasks class that has a vector of class Task

The Task class has a lot of properties. what I want to achieve is to return an array of tasks with access to the properties not only a string.

**Comment by TheLartians** [comment]
If you define the array type e.g. using `glue::createArrayClass<std::vector<int>>()` (see [here](https://github.com/TheLartians/Glue/blob/721cbb74771be98e78c2eaf2de4751cf9de7cc9f/test/source/array.cpp) for usage test-case) and add it to your `lib` object, you should be able to return it from a function and access it from TypeScript.
